pandas/util/testing.py

```diff
@@ -1247,6 +1248,10 @@ def assert_series_equal(left, right, check_dtype=True,
         right = pd.IntervalIndex(right)
         assert_index_equal(left, right, obj='{obj}.index'.format(obj=obj))
 
+    elif (is_extension_array_dtype(left) and not is_categorical_dtype(left) and
+          is_extension_array_dtype(right) and not is_categorical_dtype(right)):
+        return assert_extension_array_equal(left.values, right.values)
```

Comment:
I think we should only do this if `check_dtype=True` ?

Reply:
I'm not sure I follow your logic for this.  Before, we had no code that really compared two EA's for testing. However, that code doesn't work correctly if they are categorical dtypes.  The `check_dtype` is a separate check, so if `check_dtype` was False, you'd still do this test the same way.